src/transformers/models/audio_spectrogram_transformer/modeling_audio_spectrogram_transformer.py

```diff
@@ -234,6 +266,13 @@ def forward(
         return outputs
 
 
+# Copied from transformers.models.vit.modeling_vit.ViTSdpaAttention with ViT->AST
+class ASTSdpaAttention(ASTAttention):
+    def __init__(self, config: ASTConfig) -> None:
+        super().__init__(config)
+        self.attention = ASTSdpaSelfAttention(config)
+
+
```

Comment:
```suggestion
```

Reply:
Why do you suggest to remove sdpa attention class?

Reply:
I just don't understand why you need a dummy wrapper here

Reply:
If you need the model to use `ASTSdpaAttention` instead of `ASTSdpaSelfAttention` but copied from is an issue, you can always do something like `# Copied from transformers.models.vit.modeling_vit.ASTSdpaAttention with ASTSdpaAttention->ASTSdpaSelfAttention` something like that

Reply:
My view is like that: originally in ViT implementation there are two classes ViTSelfAttention and ViTAttention, and I have to change both of them with sdpa, I see no way to leave only one.

Reply:
@ArthurZucker When you have time, could you please clarify my understanding? I'd like to wrap it up

Reply:
Answered in the next comment, probably my bad, but if we can use `ASTSdpaSelfAttention` for both let's do it, otherwise let's go with your version!